crates/trigger-http/src/server.rs

```diff
@@ -138,12 +148,43 @@ impl<F: RuntimeFactors> HttpServer<F> {
 
     /// Serve incoming requests over the provided [`TcpListener`].
     pub async fn serve(self: Arc<Self>) -> anyhow::Result<()> {
-        let listener = TcpListener::bind(self.listen_addr).await.with_context(|| {
-            format!(
-                "Unable to listen on {listen_addr}",
-                listen_addr = self.listen_addr
-            )
-        })?;
+        let listener: TcpListener = match TcpListener::bind(self.listen_addr).await {
+            Ok(listener) => listener,
+            Err(err) => {
+                if self.find_free_port && err.kind() == ErrorKind::AddrInUse {
+                    let mut found_listener = None;
+                    for _ in 1..=9 {
+                        let mut addr = self.listen_addr;
+                        addr.set_port(addr.port() + 1);
```

Comment:
What happens if `addr.port()` is close to `U16_MAX`?  Will this overflow round to 0?  Is that expected behaviour?

Reply:
I don't think this check addresses the concern.  Look at line 158, which happens _before_ this check.  What happens if `addr.port()` returns `U16::MAX`?  What value gets passed to `set_port` in that case?

Also, `(u16::MAX + 1) as u32` is 0 in release builds and panic in debug builds, neither of which feels like what you intend...?

Reply:
Yeah, that makes sense.